odoo/addons/base/models/ir_qweb.py

```diff
@@ -1553,26 +1609,20 @@ def _compile_directive_set(self, el, compile_context, level):
                     self._compile_directive(el, compile_context, 'inner-content', 1) +
                     self._flush_text(compile_context, 1))
                 if content:
-                    def_name = self._make_name('t_set', compile_context)
-                    compile_context['_all_def'].append([f"def {def_name}(self, values, gen0, log):"] + content)
-                    expr = f"Markup(''.join({def_name}(self, values, gen0, log)))"
+                    def_name = compile_context['make_name']('t_set')
+                    compile_context['template_functions'][def_name] = [f"def {def_name}(self, values):"] + content
+                    code.append(indent_code(f"""
+                            t_set = []
+                            for item in {def_name}(self, values):
+                                if isinstance(item, tuple):
+                                    t_set.extend(item[1])
```

Comment:
to check:
- is this part tested?
- can we have a comment to remember what `item[1]` is, and why we're doing this?

also unpack `item` for readability

Reply:
II added tests. There was one of the cases that was non-functional. I fixed it.